src/components/Menu/index.js

```diff
@@ -23,37 +23,25 @@ function Menu() {
           <Link to="/visits">All Visits</Link>
         </li>
         <li>
-          <Link to="/visits/details/test-id">View a single Visit</Link>
+          <Link to="/visits/details/:id">View a single Visit</Link>
```

Comment:
When we use a link, we use an instance of the url. The dynamic url as it is defined in Routes (and in breadcrumbs) is:` "/visits/details/:id"`
Possible instances of this url are:
`"/visits/details/asdrewt234234rsfsdjkj234"`
`"/visits/details/asdf"`
`"/visits/details/test-id"`

So `:id` is just a placeholder when we define the routes. When we actually use them, we need to provide an actual value to that placeholder

Reply:
so ypu are suggesting to have a url that will end up like "/details:id" ? i mean the id will be joined on the details?

Reply:
I didn't suggest changing any urls. I just said that this change needs to be reverted. 
The Link should be like this : `<Link to="/visits/details/test-id">View a single Visit</Link>`
Do you understand the difference between the configuration url `"/visits/details/:id"` and the instance url `"/visits/details/test-id"` of that configuration?

Reply:
i am not completely sure and the reason is that if you suggest keeping a placeholder then why use /test-id instead of /:id ?

Reply:
The url of the details page of a specific visit is `"/visits/details/:id"`. The layout of the page is the same for every visit that we are going to display. What changes is the data. The data are unique for each visit. 
So when we are listing the some visits urls on a table we are going to type something like this:
`<Link to="/visits/details/visit-id-1">Press here to see the details of visit 1</Link>`
`<Link to="/visits/details/visit-id-2">Press here to see the details of visit 2</Link>`
The id of the first visit is: **visit-id-1**.  We are going to use this id in order to retrieve the data of this visit from the server.
If we click on the second link we are going to be navigated to **http://localhost:3000/visits/details/visit-id-2**. The id of that visit is **visit-id-2**. 
So we notice that there is a pattern there. The part **/visits/details/** is always the same but the last part is dynamic and it is based on the visit id. 
This is why when we tell to react router that when you see `"/visits/details/:id"` display the VisitDetails component. 
Inside the VisitDetails we are going to extract the id from the url (eg "visit-id-1", "visit-id-2") and we are going to retrieve the data of that visit from the server.
So to sum up. We use `"/visits/details/:id"` on the react router (and breadcrumb) configuration in order to direct all similar urls to the VisitDetails component.
We use `<Link to="/visits/details/visit-id-2">Press here to see the details of visit 2</Link>` to navigate to the details of a specific visit. 

In this example I created a custom link `<Link to="/visits/details/test-id">View a single Visit</Link>` in order to demonstrate how the urls are going to look. If this was an actual id (that would exist on the server) it would display the details of the visit with _id = test-id_.
Saying something like `<Link to="/visits/details/:id">View a single Visit</Link>` doesn't fit here because that is a configuration url that should be specified at the react-router configuration

Reply:
So if i get this right, the **`/visits/details/:id`** doesnt work because there is no :id page. the id is supposed to FIT to an already declared page and change it's content (because of the different data) Right?

Reply:
Exactly :id is just a placeholder that is used during the configuration of the routing